src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -223,7 +227,7 @@ public String getResourceURI(IResource resource) {
 		if(ResourceKind.LIST.equals(kind)) throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
 		try {
 			namespace = ResourceKind.PROJECT.equals(kind) ? "" : namespace;
```

Comment:
I see the same logic (use "" for namespace if we're about to build the url for a project) being repeated several times. Shouldnt we thus extract it to a single, reusable method (so that it gets easier to correct in a single location)?